src/llama.cpp

```diff
@@ -18992,10 +18993,12 @@ struct llama_context * llama_new_context_with_model(
         params.flash_attn = false;
     }
 
+#ifdef GGML_USE_METAL
     if (params.flash_attn && model->hparams.attn_soft_cap) {
         LLAMA_LOG_WARN("%s: flash_attn is not compatible with attn_soft_cap - forcing off\n", __func__);
         params.flash_attn = false;
     }
+#endif // GGML_USE_METAL
```

Comment:
We shouldn't add exceptions for specific backends like this. Eventually backends will be loaded at runtime, and these macros will be removed. It is also not completely correct, because it is possible to use `-ngl 0` with metal.

Reply:
Is it with the current code possible to do this correctly in a simple way? With a debugger I can see that `model->bufs` is already set and (I think) lets you make conclusions about the available backends. But I don't see a way in `ggml-backend.h` for converting `ggml_backend_buffer_t` to `ggml_backend_t` which I could then feed to the functions to check which backends are in use.

Alternatively, if there was a Metal implementation then this issue would also be resolved. Has there been any progress on that front?

Reply:
Just remove the check entirely and let's merge this as is, I will open a PR later with a Metal implementation. If the `supports_op` implementation is correct, it will fallback to CPU, but it will not crash.